APM > Library Injection > Add .NET instructions (#17400)
DataDog/documentation: content/en/tracing/trace_collection/library_injection_local.md

| Java       | [gcr.io/datadoghq/dd-lib-java-init][4]   | [hub.docker.com/r/datadog/dd-lib-java-init][5]   | [gallery.ecr.aws/datadog/dd-lib-java-init][6]   |
| JavaScript | [gcr.io/datadoghq/dd-lib-js-init][7]     | [hub.docker.com/r/datadog/dd-lib-js-init][8]     | [gallery.ecr.aws/datadog/dd-lib-js-init][9]     |
| Python     | [gcr.io/datadoghq/dd-lib-python-init][10] | [hub.docker.com/r/datadog/dd-lib-python-init][11] | [gallery.ecr.aws/datadog/dd-lib-python-init][12] |
| .NET       | [gcr.io/datadoghq/dd-lib-dotnet-init][13] | [hub.docker.com/r/datadog/dd-lib-dotnet-init][14] | [gallery.ecr.aws/datadog/dd-lib-dotnet-init][15] |

> | Ruby       | [gcr.io/datadoghq/dd-lib-ruby-init][13] | [hub.docker.com/r/datadog/dd-lib-ruby-init][14] | [gallery.ecr.aws/datadog/dd-lib-dotnet-init][15] |

Reviewer: ```suggestion
| Ruby       | [gcr.io/datadoghq/dd-lib-ruby-init][23] | [hub.docker.com/r/datadog/dd-lib-ruby-init][24] | [gallery.ecr.aws/datadog/dd-lib-ruby-init][25] |
```
Links need to go to Ruby

| Java       | `admission.datadoghq.com/java-lib.version: "<CONTAINER IMAGE TAG>"`   |
| JavaScript | `admission.datadoghq.com/js-lib.version: "<CONTAINER IMAGE TAG>"`     |
| Python     | `admission.datadoghq.com/python-lib.version: "<CONTAINER IMAGE TAG>"` |
| .NET       | `admission.datadoghq.com/dotnet-lib.version: "<CONTAINER IMAGE TAG>"` |

The available library versions are listed in each container registry, as well as in the tracer source repositories for each language:
- [Java][16]
- [Javascript][17]
- [Python][18]
- [.NET][19]

>   - Note: If the application container uses a musl-based Linux distribution (for example, Alpine), you must specify a tag with the the `-musl` suffix for the pod annotation.

Reviewer 2: ```suggestion
  - Note: If the application container uses a musl-based Linux distribution (such as Alpine), you must specify a tag with the the `-musl` suffix for the pod annotation. For example, if you would like to use library version `v2.29.9`, specify container tag `v2.29.9-musl`.
```

Author: Thank you for the suggestion!

| Java       | `admission.datadoghq.com/java-lib.version: "<CONTAINER IMAGE TAG>"`   |
| JavaScript | `admission.datadoghq.com/js-lib.version: "<CONTAINER IMAGE TAG>"`     |
| Python     | `admission.datadoghq.com/python-lib.version: "<CONTAINER IMAGE TAG>"` |
| .NET       | `admission.datadoghq.com/dotnet-lib.version: "<CONTAINER IMAGE TAG>"` |
| Ruby       | `admission.datadoghq.com/ruby-lib.version: "<CONTAINER IMAGE TAG>"` |

The available library versions are listed in each container registry, as well as in the tracer source repositories for each language:
- [Java][16]
- [Javascript][17]
- [Python][18]
- [.NET][19]
- [Ruby][20]

>   - Note: If the application container uses a musl-based Linux distribution (such as Alpine), you must specify a tag with the the `-musl` suffix for the pod annotation. For example, if you would like to use library version `v2.29.0`, specify container tag `v2.29.0-musl`.

Reviewer: ```suggestion
  - **Note**: If the application container uses a musl-based Linux distribution (such as Alpine), you must specify a tag with the the `-musl` suffix for the pod annotation. For example, to use library version `v2.29.0`, specify container tag `v2.29.0-musl`.
```

Author: This is a .NET only note. I've moved it back to the .NET indentation, added your suggestions, and tried to make it more clear that it's .NET only. What do you think?

**Note**: If you already have an application instrumented using version X of the library, and then use library injection to instrument using version Y of the same tracer library, the tracer does not break. Rather, the library version loaded first is used. Because library injection happens at the admission controller level prior to runtime, it takes precedent over manually configured libraries.

> **Note**: For Ruby, only support instrumenting Ruby on Rails or Hanami application.

Reviewer: ```suggestion
**Note**: For Ruby, instrumenting only Ruby on Rails or Hanami applications is supported.
```

Author: Small note: Could this go in a sub-bullet, kind of like the .NET one? Otherwise the suggested copy looks good to me

Reviewer: yeah, I'll update your branch with that
